### PR TITLE
Added barbarian assault plugin

### PR DESCRIPTION
**Pull request by Noremac201**
Add's a timer to next change, and allows left click calls. 
deepinscreenshot_select-area_20180111221914
deepinscreenshot_select-area_20180111221702

### CODE REVIEW

File: runelite-api/src/main/java/net/runelite/api/MenuAction.java [hunk 1]
```diff
@@ -139,7 +139,40 @@
...
+
+	/**
+	 * Menu action triggered by taking an object
+	 */
+	TAKE_OBJECT_GROUND(20),
+
+	/**
+	 * Menu actions triggered by calling for collector in Barbarian Assault.
+	 */
+	TELL_RED(33),
```
Comment (Adam-) [inline runelite-api/src/main/java/net/runelite/api/MenuAction.java]: 33, 34, 35, and 36 should just be FIRST/SECOND/THIRD/FOURTH_ITEM_ACTION. They are different per-object and they should not all be here. Although there are some others here already.

File: runelite-api/src/main/java/net/runelite/api/widgets/WidgetID.java [hunk 2]
```diff
@@ -227,4 +231,23 @@
 		static final int VENT_B_STATUS = 24;
 		static final int VENT_C_STATUS = 25;
 	}
+
+	static class BarbarianAssault
+	{
+		static class ATK
+		{
+			static final int CURRENT_WAVE = 1;
```
Comment (deathbeam) [inline runelite-api/src/main/java/net/runelite/api/widgets/WidgetID.java]: Why current wave and correct style 1 needs to be duplicated for ATK?

File: runelite-client/src/main/java/net/runelite/client/plugins/barbarianassault/BarbarianAssaultOverlay.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+			roleText.setText(String.format("00:%02d", currentRound.getTimeToChange()));
+			Rectangle spriteBounds = roleSprite.getBounds();
+			roleSprite.setHidden(true);
+			graphics.drawImage(plugin.getClockImage(), spriteBounds.x, spriteBounds.y, null);
+		}
+
+		return null;
+	}
+
+	public Round getCurrentRound()
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/barbarianassault/BarbarianAssaultOverlay.java]: Thus can be replaced with @Getter and @Setter annotation from Lombok on the currentRound to reduce code.

File: runelite-client/src/main/java/net/runelite/client/plugins/barbarianassault/BarbarianAssaultPlugin.java [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.FontManager;
+import net.runelite.client.ui.overlay.Overlay;
+
+@PluginDescriptor(
+	name = "Barbarian Assault Plugin"
+)
+public class BarbarianAssaultPlugin extends Plugin
+{
+	private static final List<MenuEntry> entries = new ArrayList<>();
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/barbarianassault/BarbarianAssaultPlugin.java]: Don't use static variables to hold data like this.

**Comment by tylerthardy** [comment]
@Adam- addressed this PR on Discord but not on here. 
> Adam - Today at 3:56 PM: @Noremac201 please dont add these content-specific menu items to MenuAction

 #388 should resolve any issues with using `client.setMenuEntries` - likely the reason you added the BA MenuActions to the enumeration. Cheers!

**Pull request merged**